core-library/src/main/java/org/silverpeas/core/contribution/contentcontainer/content/AbstractContentInterface.java

```diff
@@ -0,0 +1,473 @@
+    final SilverContentVisibility scv = computeSilverContentVisibility(contribution);
+    scv.setVisibilityAttributes(visible);
+    return getContentManager().addSilverContent(connection, contributionId.getLocalId(),
+        contributionId.getComponentInstanceId(), userId, scv);
+  }
+
+  /**
+   * Same as {@link #createSilverContent(Contribution, String)}, but here the {@link Connection}
+   * is given and instead of {@link Contribution} resource identifier and component instance
+   * identifier are given.<br/>
+   * Internal method {@link #computeSilverContentVisibility(Contribution)} will be called with a
+   * null parameter.
+   * @throws ContentManagerException on technical error.
+   */
+  public int createSilverContent(final Connection connection, final String resourceId,
+      final String componentInstanceId, final String userId) throws ContentManagerException {
+    final SilverContentVisibility scv = computeSilverContentVisibility(null);
+    return getContentManager()
+        .addSilverContent(connection, resourceId, componentInstanceId, userId, scv);
+  }
+
+  /**
+   * Updates the visibility of the given contribution.
+   * @param contribution the content.
+   */
+  public <T extends Contribution> void updateSilverContentVisibility(T contribution)
+      throws ContentManagerException {
+    int silverContentId = getSilverContentId(contribution);
+    if (silverContentId == -1) {
+      createSilverContent(null, contribution, contribution.getLastModifier().getId());
+    } else {
+      SilverContentVisibility scv = computeSilverContentVisibility(contribution);
+      getContentManager().updateSilverContentVisibilityAttributes(scv, silverContentId);
+    }
+  }
+
+  /**
+   * Update the visibility of the given contribution by explicitly giving it.
+   * @param contribution the content
+   * @param visibility forces the visibility so that it bypasses the one set by {@link
+   * #computeSilverContentVisibility(Contribution)}. Only taken into account on a real update.
+   */
+  public <T extends Contribution> void updateSilverContentVisibility(T contribution,
+      boolean visibility) throws ContentManagerException {
+    int silverContentId = getSilverContentId(contribution);
+    if (silverContentId == -1) {
+      createSilverContent(null, contribution, contribution.getLastModifier().getId(), visibility);
+    } else {
+      SilverContentVisibility scv = computeSilverContentVisibility(contribution);
+      scv.setVisibilityAttributes(visibility);
+      getContentManager().updateSilverContentVisibilityAttributes(scv, silverContentId);
+    }
+  }
+
+  /**
+   * Gets the default content visibility of the given contribution. Useful into creation process.
+   * @param contribution a contribution.
+   * @return a {@link SilverContentVisibility} instance initialized with default data.
+   */
+  protected <T extends Contribution> SilverContentVisibility computeSilverContentVisibility(
+      final T contribution) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "contribution". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1172)